qiita_db/util.py

```diff
@@ -2673,3 +2680,238 @@ def _resource_allocation_failures(df, k, a, b, model, col_name, type_):
     df[f'c{type_}'] = model(x_plot, k, a, b)
     failures_df = df[df[type_] > df[f'c{type_}']]
     return failures_df
+
+
+def MaxRSS_helper(x):
+    if x[-1] == 'K':
+        y = float(x[:-1]) * 1000
+    elif x[-1] == 'M':
+        y = float(x[:-1]) * 1000000
+    elif x[-1] == 'G':
+        y = float(x[:-1]) * 1000000000
+    else:
+        y = float(x)
+    return y
+
+
+def update_resource_allocation_table(test=None):
+    # Thu, Apr 27, 2023 old allocations (from barnacle) were changed to a
+    # better allocation so we default start time 2023-04-28 to
+    # use the latests for the newest version
+    """
+        Updates qiita.slurm_resource_allocation SQL table with jobs from slurm.
+        Retrieves the most recent job available in the table and appends with
+        the data.
+
+        Parameters:
+        ----------
+        test: pandas.DataFrame, optional
+            Represents dataframe containing slurm data from 2023-04-28. Used
+            for testing only.
+    """
+
+    # retrieve the most recent timestamp
+    sql_timestamp = """
+        SELECT
+            sra.job_start
+        FROM
+            qiita.slurm_resource_allocations sra
+        ORDER BY job_start DESC;
+    """
+
+    dates = ['', '']
+
+    with qdb.sql_connection.TRN:
+        sql = sql_timestamp
+        qdb.sql_connection.TRN.add(sql)
+        res = qdb.sql_connection.TRN.execute_fetchindex()
+        columns = ['timestamp']
+        timestamps = pd.DataFrame(res, columns=columns)
+        if len(timestamps['timestamp']) == 0:
+            dates[0] = datetime.strptime('2023-04-28', '%Y-%m-%d')
+        else:
+            dates[0] = str(timestamps['timestamp'].iloc[0]).split(' ')[0]
+        date0 = datetime.strptime(dates[0], '%Y-%m-%d')
+        date1 = date0 + timedelta(weeks=1)
+        dates[1] = date1.strftime('%Y-%m-%d')
+
+    sql_command = """
+            SELECT
+                pj.processing_job_id AS processing_job_id,
+                pj.external_job_id AS external_job_id
+            FROM
+                qiita.software_command sc
+            JOIN
+                qiita.processing_job pj ON pj.command_id = sc.command_id
+            JOIN
+                qiita.processing_job_status pjs
+                ON pj.processing_job_status_id = pjs.processing_job_status_id
+            LEFT JOIN
+                qiita.slurm_resource_allocations sra
+                ON pj.processing_job_id = sra.processing_job_id
+            WHERE
+                pjs.processing_job_status = 'success'
+            AND
+                pj.external_job_id ~ '^[0-9]+$'
+            AND
+                sra.processing_job_id IS NULL;
+        """
```

Comment:
Note that this query will return _all_ the rows, which will be a lot so it will be good to limit based on the job_id of the latests result in the previous block (what you are using to extract the time). Could you limit to just return anything newer than that job?

Reply:
Would we determine whether the job is newer through external_id? I don't think qiita.processing_job stores information about when the job was submitted.

Reply:
That's correct, the external id is an incremental integer so it can be used to find what's newer.